<a href="https://colab.research.google.com/github/TheBatmanofButler/racial-struggles-project/blob/master/census_race_data_1850_1900_cleanup.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install geopandas==0.3.0
!pip install pyshp==1.2.10
!pip install shapely==1.6.3
!pip install plotly-geo

In [0]:
import pandas as pd
import plotly
import plotly.figure_factory as ff

What happened to slaves after the Civil War ended?


In [0]:
TOTAL_POP_DATA_URL = "https://raw.githubusercontent.com/TheBatmanofButler/racial-struggles-project/master/data/nhgis0002_csv/nhgis0002_ts_nominal_state.csv" 

In [0]:
total_pop = pd.read_csv(TOTAL_POP_DATA_URL)

In [5]:
total_pop.head()

GISJOIN              STATE  STATEFP  ...  A00AA1990  A00AA2000  A00AA2010
0    G010            Alabama      1.0  ...  4040587.0  4447100.0  4779736.0
1    G020             Alaska      2.0  ...   550043.0   626932.0   710231.0
2    G025   Alaska Territory      NaN  ...        NaN        NaN        NaN
3    G040            Arizona      4.0  ...  3665228.0  5130632.0  6392017.0
4    G045  Arizona Territory      NaN  ...        NaN        NaN        NaN

[5 rows x 27 columns]

In [6]:
total_pop = total_pop.drop(columns=["GISJOIN", "STATEFP", "STATENH"])
columns = total_pop.columns
print(columns)

Index(['STATE', 'A00AA1790', 'A00AA1800', 'A00AA1810', 'A00AA1820',
       'A00AA1830', 'A00AA1840', 'A00AA1850', 'A00AA1860', 'A00AA1870',
       'A00AA1880', 'A00AA1890', 'A00AA1900', 'A00AA1910', 'A00AA1920',
       'A00AA1930', 'A00AA1940', 'A00AA1950', 'A00AA1960', 'A00AA1970',
       'A00AA1980', 'A00AA1990', 'A00AA2000', 'A00AA2010'],
      dtype='object')


In [7]:
new_columns = []

for column in columns:
  if column.find("A00AA") == -1:
    new_columns.append(column)
  else:
    new_column = column[5:]
    new_columns.append(new_column)

print(new_columns)

['STATE', '1790', '1800', '1810', '1820', '1830', '1840', '1850', '1860', '1870', '1880', '1890', '1900', '1910', '1920', '1930', '1940', '1950', '1960', '1970', '1980', '1990', '2000', '2010']


In [0]:
total_pop.columns = new_columns

In [9]:
total_pop.head()

STATE  1790  1800  ...       1990       2000       2010
0            Alabama   NaN   NaN  ...  4040587.0  4447100.0  4779736.0
1             Alaska   NaN   NaN  ...   550043.0   626932.0   710231.0
2   Alaska Territory   NaN   NaN  ...        NaN        NaN        NaN
3            Arizona   NaN   NaN  ...  3665228.0  5130632.0  6392017.0
4  Arizona Territory   NaN   NaN  ...        NaN        NaN        NaN

[5 rows x 24 columns]

In [10]:
total_pop.fillna(0)

STATE  1790  1800  ...       1990       2000       2010
0                   Alabama   0.0   0.0  ...  4040587.0  4447100.0  4779736.0
1                    Alaska   0.0   0.0  ...   550043.0   626932.0   710231.0
2          Alaska Territory   0.0   0.0  ...        0.0        0.0        0.0
3                   Arizona   0.0   0.0  ...  3665228.0  5130632.0  6392017.0
4         Arizona Territory   0.0   0.0  ...        0.0        0.0        0.0
..                      ...   ...   ...  ...        ...        ...        ...
79      Wisconsin Territory   0.0   0.0  ...        0.0        0.0        0.0
80                  Wyoming   0.0   0.0  ...   453588.0   493782.0   563626.0
81        Wyoming Territory   0.0   0.0  ...        0.0        0.0        0.0
82              Puerto Rico   0.0   0.0  ...        0.0        0.0  3725789.0
83  Persons in the Military   0.0   0.0  ...        0.0        0.0        0.0

[84 rows x 24 columns]

In [0]:
RACIAL_DATA_1850_URL = "https://raw.githubusercontent.com/TheBatmanofButler/racial-struggles-project/master/data/nhgis0003_csv/nhgis0003_ds10_1850_state.csv"

In [0]:
racial_data_1850 = pd.read_csv(RACIAL_DATA_1850_URL)

In [13]:
racial_data_1850.head()

GISJOIN  YEAR        STATE  STATEA  ...     AREANAME  AE6001  AE6002  AE6003
0    G010  1850      Alabama      10  ...      Alabama  426514    2265  342844
1    G050  1850     Arkansas      50  ...     Arkansas  162189     608   47100
2    G060  1850   California      60  ...   California   91635     962       0
3    G090  1850  Connecticut      90  ...  Connecticut  363099    7693       0
4    G100  1850     Delaware     100  ...     Delaware   71169   18073    2290

[5 rows x 9 columns]

In [14]:
racial_data_1850 = racial_data_1850.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1850.columns
print(columns)

Index(['YEAR', 'STATE', 'AE6001', 'AE6002', 'AE6003'], dtype='object')


In [0]:
racial_data_1850 = racial_data_1850.rename(columns={"AE6001": "White", "AE6002": "Nonwhite: Free", "AE6003": "Nonwhite: Slave"})

In [16]:
racial_data_1850.head()

YEAR        STATE   White  Nonwhite: Free  Nonwhite: Slave
0  1850      Alabama  426514            2265           342844
1  1850     Arkansas  162189             608            47100
2  1850   California   91635             962                0
3  1850  Connecticut  363099            7693                0
4  1850     Delaware   71169           18073             2290

In [0]:
RACIAL_DATA_1860_URL = "https://raw.githubusercontent.com/TheBatmanofButler/racial-struggles-project/master/data/nhgis0003_csv/nhgis0003_ds14_1860_state.csv"

In [0]:
racial_data_1860 = pd.read_csv(RACIAL_DATA_1860_URL)

In [19]:
racial_data_1860.head()

GISJOIN  YEAR               STATE  STATEA  ...  AH3003   AH3004  AH3005   AH3006
0    G010  1860             Alabama      10  ...  435080    160.0     NaN      NaN
1    G050  1860            Arkansas      50  ...  111115     48.0     NaN      NaN
2    G060  1860          California      60  ...       0  17798.0     NaN  34933.0
3    G085  1860  Colorado Territory      85  ...       0      0.0     NaN      NaN
4    G090  1860         Connecticut      90  ...       0     16.0     NaN      NaN

[5 rows x 12 columns]

In [20]:
racial_data_1860 = racial_data_1860.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1860.columns
print(columns)

Index(['YEAR', 'STATE', 'AH3001', 'AH3002', 'AH3003', 'AH3004', 'AH3005',
       'AH3006'],
      dtype='object')


In [0]:
racial_data_1860 = racial_data_1860.rename(columns={"AH3001": "White", "AH3002": "Free colored", "AH3003": "Slave", "AH3004": "Indian", "AH3005": "Half breed", "AH3006": "Asiatic"})

In [22]:
racial_data_1860.head()

YEAR               STATE   White  ...   Indian  Half breed  Asiatic
0  1860             Alabama  526271  ...    160.0         NaN      NaN
1  1860            Arkansas  324143  ...     48.0         NaN      NaN
2  1860          California  323177  ...  17798.0         NaN  34933.0
3  1860  Colorado Territory   34231  ...      0.0         NaN      NaN
4  1860         Connecticut  451504  ...     16.0         NaN      NaN

[5 rows x 8 columns]

In [0]:
racial_data_territories_1850 = racial_data_1850[racial_data_1850["STATE"].str.contains("Territory")]

In [24]:
racial_data_territories_1850

YEAR                 STATE  White  Nonwhite: Free  Nonwhite: Slave
17  1850   Minnesota Territory   6038              39                0
22  1850  New Mexico Territory  61525              22                0
26  1850      Oregon Territory  13087             207                0
32  1850        Utah Territory  11330              24               26

In [0]:
racial_data_territories_1860 = racial_data_1860[racial_data_1860["STATE"].str.contains("Territory")]

In [26]:
racial_data_territories_1860

YEAR                 STATE   White  ...   Indian  Half breed  Asiatic
3   1860    Colorado Territory   34231  ...      0.0         NaN      NaN
5   1860      Dakota Territory    2576  ...   2261.0         NaN      NaN
13  1860      Kansas Territory  106390  ...    189.0         NaN      NaN
23  1860    Nebraska Territory   28696  ...     63.0         NaN      NaN
24  1860      Nevada Territory    6812  ...      0.0         NaN      NaN
27  1860  New Mexico Territory   82924  ...  10452.0        55.0      NaN
37  1860        Utah Territory   40125  ...     89.0         NaN      NaN
40  1860  Washington Territory   11138  ...    426.0         NaN      NaN

[8 rows x 8 columns]

There's not that many slaves in the territories, so gonna just ignore those areas.

In [0]:
racial_data_1850 = racial_data_1850[~racial_data_1850["STATE"].str.contains("Territory")]

In [28]:
racial_data_1850

YEAR                 STATE    White  Nonwhite: Free  Nonwhite: Slave
0   1850               Alabama   426514            2265           342844
1   1850              Arkansas   162189             608            47100
2   1850            California    91635             962                0
3   1850           Connecticut   363099            7693                0
4   1850              Delaware    71169           18073             2290
5   1850  District Of Columbia    37941           10059             3687
6   1850               Florida    47203             932            39310
7   1850               Georgia   521572            2931           381682
8   1850              Illinois   846034            5436                0
9   1850               Indiana   977154           11262                0
10  1850                  Iowa   191881             333                0
11  1850              Kentucky   761413           10011           210981
12  1850             Louisiana   255491           17462           244809
13  1850                 Maine   581813            1356                0
14  1850              Maryland   417943           74723            90368
15  1850         Massachusetts   985450            9064                0
16  1850              Michigan   395071            2583                0
18  1850           Mississippi   295718             930           309878
19  1850              Missouri   592004            2618            87422
20  1850         New Hampshire   317456             520                0
21  1850            New Jersey   465509           23810              236
23  1850              New York  3048325           49069                0
24  1850        North Carolina   553028           27463           288548
25  1850                  Ohio  1955050           25279                0
27  1850          Pennsylvania  2258160           53626                0
28  1850          Rhode Island   143875            3670                0
29  1850        South Carolina   274563            8960           384984
30  1850             Tennessee   756836            6422           239459
31  1850                 Texas   154034             397            58161
33  1850               Vermont   313402             718                0
34  1850              Virginia   894800           54333           472528
35  1850             Wisconsin   304756             635                0

In [0]:
racial_data_1860 = racial_data_1860[~racial_data_1860["STATE"].str.contains("Territory")]

In [30]:
racial_data_1860

YEAR                 STATE    White  ...   Indian  Half breed  Asiatic
0   1860               Alabama   526271  ...    160.0         NaN      NaN
1   1860              Arkansas   324143  ...     48.0         NaN      NaN
2   1860            California   323177  ...  17798.0         NaN  34933.0
4   1860           Connecticut   451504  ...     16.0         NaN      NaN
6   1860              Delaware    90589  ...      0.0         NaN      NaN
7   1860  District Of Columbia    60763  ...      1.0         NaN      NaN
8   1860               Florida    77746  ...      1.0         NaN      NaN
9   1860               Georgia   591550  ...     38.0         NaN      NaN
10  1860              Illinois  1704291  ...     32.0         NaN      NaN
11  1860               Indiana  1338710  ...    290.0         NaN      NaN
12  1860                  Iowa   673779  ...     65.0         NaN      NaN
14  1860              Kentucky   919484  ...     33.0         NaN      NaN
15  1860             Louisiana   357456  ...    173.0         NaN      NaN
16  1860                 Maine   626947  ...      5.0         NaN      NaN
17  1860              Maryland   515918  ...      0.0         NaN      NaN
18  1860         Massachusetts  1221432  ...     32.0         NaN      NaN
19  1860              Michigan   736142  ...   6172.0         NaN      NaN
20  1860             Minnesota   169395  ...   2369.0         NaN      NaN
21  1860           Mississippi   353899  ...      2.0         NaN      NaN
22  1860              Missouri  1063489  ...     20.0         NaN      NaN
25  1860         New Hampshire   325579  ...      NaN         NaN      NaN
26  1860            New Jersey   646699  ...      0.0         NaN      NaN
28  1860              New York  3831590  ...    140.0         NaN      NaN
29  1860        North Carolina   629942  ...   1158.0         NaN      NaN
30  1860                  Ohio  2302808  ...     30.0         NaN      NaN
31  1860                Oregon    52160  ...    177.0         NaN      NaN
32  1860          Pennsylvania  2849259  ...      7.0         NaN      NaN
33  1860          Rhode Island   170649  ...     19.0         NaN      NaN
34  1860        South Carolina   291300  ...     88.0         NaN      NaN
35  1860             Tennessee   826722  ...     60.0         NaN      NaN
36  1860                 Texas   420891  ...    403.0         NaN      NaN
38  1860               Vermont   314369  ...     20.0         NaN      NaN
39  1860              Virginia  1047299  ...    112.0         NaN      NaN
41  1860             Wisconsin   773693  ...    613.0       404.0      NaN

[34 rows x 8 columns]

In [0]:
new_states_1860 = set(racial_data_1860["STATE"]) - set(racial_data_1850["STATE"])

In [0]:
racial_data_new_states_1860 = racial_data_1860[racial_data_1860["STATE"].str.contains("|".join(new_states_1860))]


In [33]:
racial_data_new_states_1860

YEAR      STATE   White  Free colored  Slave  Indian  Half breed  Asiatic
20  1860  Minnesota  169395           259      0  2369.0         NaN      NaN
31  1860     Oregon   52160           128      0   177.0         NaN      NaN

In [0]:
RACIAL_DATA_1870_URL = "https://raw.githubusercontent.com/TheBatmanofButler/racial-struggles-project/master/data/nhgis0005_csv/nhgis0005_ds17_1870_state.csv"

In [0]:
racial_data_1870 = pd.read_csv(RACIAL_DATA_1870_URL)

In [36]:
racial_data_1870.head()

GISJOIN  YEAR               STATE  STATEA  ...  AK3001  AK3002  AK3003  AK3004
0    G010  1870             Alabama      10  ...  521384  475510       0      98
1    G045  1870   Arizona Territory      45  ...    9581      26      20      31
2    G050  1870            Arkansas      50  ...  362115  122169      98      89
3    G060  1870          California      60  ...  499424    4272   49310    7241
4    G085  1870  Colorado Territory      85  ...   39221     456       7     180

[5 rows x 10 columns]

In [37]:
racial_data_1870 = racial_data_1870.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1870.columns
print(columns)

Index(['YEAR', 'STATE', 'AK3001', 'AK3002', 'AK3003', 'AK3004'], dtype='object')


In [0]:
racial_data_1870 = racial_data_1870.rename(columns={"AK3001": "White", "AK3002": "Colored", "AK3003": "Chinese", "AK3004": "Indian"})

In [39]:
racial_data_1870.head()

YEAR               STATE   White  Colored  Chinese  Indian
0  1870             Alabama  521384   475510        0      98
1  1870   Arizona Territory    9581       26       20      31
2  1870            Arkansas  362115   122169       98      89
3  1870          California  499424     4272    49310    7241
4  1870  Colorado Territory   39221      456        7     180

In [0]:
racial_data_territories_1870 = racial_data_1870[racial_data_1870["STATE"].str.contains("Territory")]

In [41]:
racial_data_territories_1870

YEAR                 STATE  White  Colored  Chinese  Indian
1   1870     Arizona Territory   9581       26       20      31
4   1870    Colorado Territory  39221      456        7     180
6   1870      Dakota Territory  12887       94        0    1200
11  1870       Idaho Territory  10618       60     4274      47
25  1870     Montana Territory  18306      183     1949     157
30  1870  New Mexico Territory  90393      172        0    1309
40  1870        Utah Territory  86044      118      445     179
43  1870  Washington Territory  22195      207      234    1319
46  1870     Wyoming Territory   8726      183      143      66

Ignoring territories again. Highest "colored" percentage in any territory is ~2% in Wyoming.

In [0]:
racial_data_1870 = racial_data_1870[~racial_data_1870["STATE"].str.contains("Territory")]

In [43]:
racial_data_1870

YEAR                 STATE    White  Colored  Chinese  Indian
0   1870               Alabama   521384   475510        0      98
2   1870              Arkansas   362115   122169       98      89
3   1870            California   499424     4272    49310    7241
5   1870           Connecticut   527549     9668        2     235
7   1870              Delaware   102221    22794        0       0
8   1870  District Of Columbia    88278    43404        3      15
9   1870               Florida    96057    91689        0       2
10  1870               Georgia   638926   545142        1      40
12  1870              Illinois  2511096    28762        1      32
13  1870               Indiana  1655837    24560        0     240
14  1870                  Iowa  1188207     5762        3      48
15  1870                Kansas   346377    17108        0     914
16  1870              Kentucky  1098692   222210        1     108
17  1870             Louisiana   362065   364210       71     569
18  1870                 Maine   624809     1606        1     499
19  1870              Maryland   605497   175391        2       4
20  1870         Massachusetts  1443156    13947       97     151
21  1870              Michigan  1167282    11849        2    4926
22  1870             Minnesota   438257      759        0     690
23  1870           Mississippi   382896   444201       16     809
24  1870              Missouri  1603146   118071        3      75
26  1870              Nebraska   122117      789        0      87
27  1870                Nevada    38959      357     3152      23
28  1870         New Hampshire   317697      580        0      23
29  1870            New Jersey   875407    30658       15      16
31  1870              New York  4330210    52081       29     439
32  1870        North Carolina   678470   391650        0    1241
33  1870                  Ohio  2601946    63213        1     100
34  1870                Oregon    86929      346     3330     318
35  1870          Pennsylvania  3456609    65294       14      34
36  1870          Rhode Island   212219     4980        0     154
37  1870        South Carolina   289667   415814        1     124
38  1870             Tennessee   936119   322331        0      70
39  1870                 Texas   564700   253475       25     379
41  1870               Vermont   329613      924        0      14
42  1870              Virginia   712089   512841        4     229
44  1870         West Virginia   424033    17980        0       1
45  1870             Wisconsin  1051351     2113        0    1206

In [0]:
RACIAL_DATA_1880_URL = "https://raw.githubusercontent.com/TheBatmanofButler/racial-struggles-project/master/data/nhgis0005_csv/nhgis0005_ds23_1880_state.csv"

In [0]:
racial_data_1880 = pd.read_csv(RACIAL_DATA_1880_URL)

In [46]:
racial_data_1880.head()

GISJOIN  YEAR              STATE  STATEA  ...  APP001  APP002  APP003  APP004
0    G010  1880            Alabama      10  ...  662185  600103       4     213
1    G025  1880   Alaska Territory      25  ...     430       0       0   32996
2    G045  1880  Arizona Territory      45  ...   35160     155    1632    3493
3    G050  1880           Arkansas      50  ...  591531  210666     133     195
4    G060  1880         California      60  ...  767181    6018   75218   16277

[5 rows x 10 columns]

In [47]:
racial_data_1880 = racial_data_1880.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1880.columns
print(columns)

Index(['YEAR', 'STATE', 'APP001', 'APP002', 'APP003', 'APP004'], dtype='object')


In [0]:
racial_data_1880 = racial_data_1880.rename(columns={"APP001": "White", "APP002": "Colored", "APP003": "Chinese", "APP004": "Indian"})

In [49]:
racial_data_1880.head()

YEAR              STATE   White  Colored  Chinese  Indian
0  1880            Alabama  662185   600103        4     213
1  1880   Alaska Territory     430        0        0   32996
2  1880  Arizona Territory   35160      155     1632    3493
3  1880           Arkansas  591531   210666      133     195
4  1880         California  767181     6018    75218   16277

In [0]:
racial_data_territories_1880 = racial_data_1880[racial_data_1880["STATE"].str.contains("Territory")]

In [51]:
racial_data_territories_1880

YEAR                 STATE   White  Colored  Chinese  Indian
1   1880      Alaska Territory     430        0        0   32996
2   1880     Arizona Territory   35160      155     1632    3493
7   1880      Dakota Territory  133147      401      238    1391
12  1880       Idaho Territory   29013       53     3379     165
26  1880     Montana Territory   35385      346     1765    1663
31  1880  New Mexico Territory  108721     1015       57    9772
41  1880        Utah Territory  142423      232      501     807
44  1880  Washington Territory   67199      325     3187    4405
47  1880     Wyoming Territory   19437      298      914     140

Ignoring territories again.

In [0]:
racial_data_1880 = racial_data_1880[~racial_data_1880["STATE"].str.contains("Territory")]

In [53]:
racial_data_1880

YEAR                 STATE    White  Colored  Chinese  Indian
0   1880               Alabama   662185   600103        4     213
3   1880              Arkansas   591531   210666      133     195
4   1880            California   767181     6018    75218   16277
5   1880              Colorado   191126     2435      612     154
6   1880           Connecticut   610769    11547      129     255
8   1880              Delaware   120160    26442        1       5
9   1880  District Of Columbia   118006    59596       17       5
10  1880               Florida   142605   126690       18     180
11  1880               Georgia   816906   725133       17     124
13  1880              Illinois  3031151    46368      212     140
14  1880               Indiana  1938798    39228       29     246
15  1880                  Iowa  1614600     9516       33     466
16  1880                Kansas   952155    43107       19     815
17  1880              Kentucky  1377179   271451       10      50
18  1880             Louisiana   454954   483655      489     848
19  1880                 Maine   646852     1451        8     625
20  1880              Maryland   724693   210230        5      15
21  1880         Massachusetts  1763782    18697      237     369
22  1880              Michigan  1614560    15100       28    7249
23  1880             Minnesota   776884     1564       25    2300
24  1880           Mississippi   479398   650291       51    1857
25  1880              Missouri  2022826   145350       91     113
27  1880              Nebraska   449764     2385       18     235
28  1880                Nevada    53556      488     5419    2803
29  1880         New Hampshire   346229      685       14      63
30  1880            New Jersey  1092017    38853      172      74
32  1880              New York  5016022    65104      926     819
33  1880        North Carolina   867242   531277        1    1230
34  1880                  Ohio  3117920    79900      112     130
35  1880                Oregon   163075      487     9512    1694
36  1880          Pennsylvania  4197016    85535      156     184
37  1880          Rhode Island   269939     6488       27      77
38  1880        South Carolina   391105   604332        9     131
39  1880             Tennessee  1138831   403151       25     352
40  1880                 Texas  1197237   393384      136     992
42  1880               Vermont   331218     1057        0      11
43  1880              Virginia   880858   631616        6      85
45  1880         West Virginia   592537    25886        5      29
46  1880             Wisconsin  1309618     2702       16    3161

1890 race data is split into White and non-White

In [0]:
RACIAL_DATA_WHITE_1890_URL = "https://github.com/TheBatmanofButler/racial-struggles-project/raw/master/data/nhgis0005_csv/nhgis0005_ds27_1890_state.csv"

In [0]:
racial_data_white_1890 = pd.read_csv(RACIAL_DATA_WHITE_1890_URL)

In [56]:
racial_data_white_1890.head()

GISJOIN  YEAR              STATE  ...  COUNTYA           AREANAME     AV2001
0    G010  1890            Alabama  ...      NaN            Alabama   833718.0
1    G025  1890   Alaska Territory  ...      NaN   Alaska Territory        NaN
2    G045  1890  Arizona Territory  ...      NaN  Arizona Territory    55580.0
3    G050  1890           Arkansas  ...      NaN           Arkansas   818752.0
4    G060  1890         California  ...      NaN         California  1111672.0

[5 rows x 7 columns]

In [0]:
RACIAL_DATA_NONWHITE_1890_URL = "https://github.com/TheBatmanofButler/racial-struggles-project/raw/master/data/nhgis0005_csv/nhgis0005_ds29_1890_state.csv"

In [0]:
racial_data_nonwhite_1890 = pd.read_csv(RACIAL_DATA_NONWHITE_1890_URL)

In [59]:
racial_data_nonwhite_1890.head()

GISJOIN  YEAR              STATE  STATEA  ... AWL004  AWL005  AWL006  AWL007
0    G010  1890            Alabama      10  ...   4387      48       3     759
1    G045  1890  Arizona Territory      45  ...     15    1170       1    1512
2    G050  1890           Arkansas      50  ...   2344      92       0     218
3    G060  1890         California      60  ...   1264   72472    1147   11517
4    G080  1890           Colorado      80  ...    152    1398      10     107

[5 rows x 12 columns]

In [0]:
extra_territories_white_1890 = set(racial_data_white_1890["STATE"]) - set(racial_data_nonwhite_1890["STATE"])

In [61]:
extra_territories_white_1890

{'Alaska Territory', 'Indian Territory'}

Ignoring those two territories since there's no nonwhite data for them

In [0]:
racial_data_white_1890 = racial_data_white_1890[~racial_data_white_1890["STATE"].str.contains("|".join(extra_territories_white_1890))]

In [63]:
racial_data_white_1890

GISJOIN  YEAR  ...              AREANAME     AV2001
0     G010  1890  ...               Alabama   833718.0
2     G045  1890  ...     Arizona Territory    55580.0
3     G050  1890  ...              Arkansas   818752.0
4     G060  1890  ...            California  1111672.0
5     G080  1890  ...              Colorado   404468.0
6     G090  1890  ...           Connecticut   733438.0
7     G100  1890  ...              Delaware   140066.0
8     G110  1890  ...  District of Columbia   154695.0
9     G120  1890  ...               Florida   224949.0
10    G130  1890  ...               Georgia   978357.0
11    G160  1890  ...                 Idaho    82018.0
12    G170  1890  ...              Illinois  3768472.0
14    G180  1890  ...               Indiana  2146736.0
15    G190  1890  ...                  Iowa  1901086.0
16    G200  1890  ...                Kansas  1376553.0
17    G210  1890  ...              Kentucky  1590462.0
18    G220  1890  ...            Louisianna   558395.0
19    G230  1890  ...                 Maine   659263.0
20    G240  1890  ...              Maryland   826493.0
21    G250  1890  ...         Massachusetts  2215373.0
22    G260  1890  ...              Michigan  2072884.0
23    G270  1890  ...             Minnesota  1296159.0
24    G280  1890  ...           Mississippi   544851.0
25    G290  1890  ...              Missouri  2528458.0
26    G300  1890  ...               Montana   127271.0
27    G310  1890  ...              Nebraska  1046888.0
28    G320  1890  ...                Nevada    39084.0
29    G330  1890  ...         New Hampshire   375840.0
30    G340  1890  ...            New Jersey  1396581.0
31    G355  1890  ...  New Mexico Territory   142719.0
32    G360  1890  ...              New York  5923952.0
33    G370  1890  ...        North Carolina  1055382.0
34    G380  1890  ...          North Dakota   182123.0
35    G390  1890  ...                  Ohio  3584805.0
36    G405  1890  ...    Oklahoma Territory    58826.0
37    G410  1890  ...                Oregon   301758.0
38    G420  1890  ...          Pennsylvania  5148257.0
39    G440  1890  ...          Rhode Island   337859.0
40    G450  1890  ...        South Carolina   462008.0
41    G460  1890  ...          South Dakota   327290.0
42    G470  1890  ...             Tennessee  1336637.0
43    G480  1890  ...                 Texas  1745935.0
44    G495  1890  ...        Utah Territory   205899.0
45    G500  1890  ...               Vermont   331418.0
46    G510  1890  ...              Virginia  1020122.0
47    G530  1890  ...            Washington   340513.0
48    G540  1890  ...         West Virginia   730077.0
49    G550  1890  ...             Wisconsin  1680473.0
50    G560  1890  ...               Wyoming    59275.0

[49 rows x 7 columns]

In [0]:
racial_data_white_1890 = racial_data_white_1890.set_index("STATE")

In [0]:
racial_data_nonwhite_1890 = racial_data_nonwhite_1890.set_index("STATE")

In [0]:
racial_data_1890 = pd.concat([racial_data_white_1890, racial_data_nonwhite_1890], axis=1)

In [67]:
racial_data_1890.head()

GISJOIN  YEAR  STATEA  ...  AWL005 AWL006  AWL007
STATE                                    ...                       
Alabama              G010  1890      10  ...      48      3     759
Arizona Territory    G045  1890      45  ...    1170      1    1512
Arkansas             G050  1890      50  ...      92      0     218
California           G060  1890      60  ...   72472   1147   11517
Colorado             G080  1890      80  ...    1398     10     107

[5 rows x 17 columns]

In [0]:
racial_data_1890 = racial_data_1890.reset_index()

In [69]:
racial_data_1890.head()

STATE GISJOIN  YEAR  STATEA  ...  AWL004 AWL005  AWL006 AWL007
0            Alabama    G010  1890      10  ...    4387     48       3    759
1  Arizona Territory    G045  1890      45  ...      15   1170       1   1512
2           Arkansas    G050  1890      50  ...    2344     92       0    218
3         California    G060  1890      60  ...    1264  72472    1147  11517
4           Colorado    G080  1890      80  ...     152   1398      10    107

[5 rows x 18 columns]

In [0]:
racial_data_1890 = racial_data_1890.loc[:,~racial_data_1890.columns.duplicated()]

In [71]:
racial_data_1890.head()

STATE GISJOIN  YEAR  STATEA  ...  AWL004 AWL005  AWL006  AWL007
0            Alabama    G010  1890      10  ...    4387     48       3     759
1  Arizona Territory    G045  1890      45  ...      15   1170       1    1512
2           Arkansas    G050  1890      50  ...    2344     92       0     218
3         California    G060  1890      60  ...    1264  72472    1147   11517
4           Colorado    G080  1890      80  ...     152   1398      10     107

[5 rows x 14 columns]

In [72]:
racial_data_1890 = racial_data_1890.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1890.columns
print(columns)

Index(['STATE', 'YEAR', 'AV2001', 'AWL001', 'AWL002', 'AWL003', 'AWL004',
       'AWL005', 'AWL006', 'AWL007'],
      dtype='object')


In [0]:
racial_data_1890 = racial_data_1890.rename(columns={
    "AV2001": "White",
    "AWL001": "Negroes",
    "AWL002": "Mulattoes",
    "AWL003": "Quadroons",
    "AWL004": "Octoroons",
    "AWL005": "Chinese",
    "AWL006": "Japanese",
    "AWL007": "Civilized Indians",
})

In [74]:
racial_data_1890.head()

STATE  YEAR      White  ...  Chinese  Japanese  Civilized Indians
0            Alabama  1890   833718.0  ...       48         3                759
1  Arizona Territory  1890    55580.0  ...     1170         1               1512
2           Arkansas  1890   818752.0  ...       92         0                218
3         California  1890  1111672.0  ...    72472      1147              11517
4           Colorado  1890   404468.0  ...     1398        10                107

[5 rows x 10 columns]

In [0]:
racial_data_territories_1890 = racial_data_1890[racial_data_1890["STATE"].str.contains("Territory")]

In [76]:
racial_data_territories_1890

STATE  YEAR     White  ...  Chinese  Japanese  Civilized Indians
1      Arizona Territory  1890   55580.0  ...     1170         1               1512
29  New Mexico Territory  1890  142719.0  ...      361         3               8554
34    Oklahoma Territory  1890   58826.0  ...       25         0                 10
42        Utah Territory  1890  205899.0  ...      806         4                608

[4 rows x 10 columns]

Ignoring territories again.

In [0]:
racial_data_1890 = racial_data_1890[~racial_data_1890["STATE"].str.contains("Territory")]

In [78]:
racial_data_1890

STATE  YEAR      White  ...  Chinese  Japanese  Civilized Indians
0                Alabama  1890   833718.0  ...       48         3                759
2               Arkansas  1890   818752.0  ...       92         0                218
3             California  1890  1111672.0  ...    72472      1147              11517
4               Colorado  1890   404468.0  ...     1398        10                107
5            Connecticut  1890   733438.0  ...      272        18                228
6               Delaware  1890   140066.0  ...       37         0                  4
7   District Of Columbia  1890   154695.0  ...       91         9                 25
8                Florida  1890   224949.0  ...      108        14                171
9                Georgia  1890   978357.0  ...      108         5                 68
10                 Idaho  1890    82018.0  ...     2007         0                159
11              Illinois  1890  3768472.0  ...      740        14                 97
12               Indiana  1890  2146736.0  ...       92        18                343
13                  Iowa  1890  1901086.0  ...       64         1                 60
14                Kansas  1890  1376553.0  ...       93         4                736
15              Kentucky  1890  1590462.0  ...       28         3                 71
16             Louisiana  1890   558395.0  ...      333        39                627
17                 Maine  1890   659263.0  ...       73         1                559
18              Maryland  1890   826493.0  ...      189         7                 44
19         Massachusetts  1890  2215373.0  ...      984        18                424
20              Michigan  1890  2072884.0  ...      120        38               5624
21             Minnesota  1890  1296159.0  ...       94         2               1888
22           Mississippi  1890   544851.0  ...      147         7               2036
23              Missouri  1890  2528458.0  ...      409         6                127
24               Montana  1890   127271.0  ...     2532         6                860
25              Nebraska  1890  1046888.0  ...      214         2               2893
26                Nevada  1890    39084.0  ...     2833         3               3599
27         New Hampshire  1890   375840.0  ...       58         2                 16
28            New Jersey  1890  1396581.0  ...      608        22                 84
30              New York  1890  5923952.0  ...     2935       148                726
31        North Carolina  1890  1055382.0  ...       32         1               1514
32          North Dakota  1890   182123.0  ...       28         1                194
33                  Ohio  1890  3584805.0  ...      183        22                193
35                Oregon  1890   301758.0  ...     9540        25               1258
36          Pennsylvania  1890  5148257.0  ...     1146        32                983
37          Rhode Island  1890   337859.0  ...       69         5                180
38        South Carolina  1890   462008.0  ...       34         0                173
39          South Dakota  1890   327290.0  ...      195         0                782
40             Tennessee  1890  1336637.0  ...       51         6                146
41                 Texas  1890  1745935.0  ...      710         3                704
43               Vermont  1890   331418.0  ...       32         1                 34
44              Virginia  1890  1020122.0  ...       55        16                349
45            Washington  1890   340513.0  ...     3260       360               3655
46         West Virginia  1890   730077.0  ...       15         3                  9
47             Wisconsin  1890  1680473.0  ...      119         9               3835
48               Wyoming  1890    59275.0  ...      465         0                 43

[45 rows x 10 columns]

In [0]:
RACIAL_DATA_1900_URL = "https://github.com/TheBatmanofButler/racial-struggles-project/raw/master/data/nhgis0005_csv/nhgis0005_ds31_1900_state.csv"

In [0]:
racial_data_1900 = pd.read_csv(RACIAL_DATA_1900_URL)

In [81]:
racial_data_1900.head()

GISJOIN  YEAR              STATE  STATEA  ...  AZ3002  AZ3003  AZ3004   AZ2001
0    G010  1900            Alabama      10  ...  418159  409237  418070  1001152
1    G025  1900   Alaska Territory      25  ...   14520     151      17    30493
2    G045  1900  Arizona Territory      45  ...   13463    1363     485    92903
3    G050  1900           Arkansas      50  ...  181542  185342  181514   944580
4    G060  1900         California      60  ...   16942    5766    5279  1402727

[5 rows x 11 columns]

In [82]:
racial_data_1900 = racial_data_1900.drop(columns=["GISJOIN", "STATEA", "COUNTYA", "AREANAME"])
columns = racial_data_1900.columns
print(columns)

Index(['YEAR', 'STATE', 'AZ3001', 'AZ3002', 'AZ3003', 'AZ3004', 'AZ2001'], dtype='object')


1900 data is split into males and females for "negro" and "other colored"

In [0]:
racial_data_1900 = racial_data_1900.rename(columns={
    "AZ2001": "White",
    "AZ3001": "Other Colored >> Male",
    "AZ3002": "Other Colored >> Female",
    "AZ3003": "Negro >> Male",
    "AZ3004": "Negro >> Female"
})

In [84]:
racial_data_1900.head()

YEAR              STATE  ...  Negro >> Female    White
0  1900            Alabama  ...           418070  1001152
1  1900   Alaska Territory  ...               17    30493
2  1900  Arizona Territory  ...              485    92903
3  1900           Arkansas  ...           181514   944580
4  1900         California  ...             5279  1402727

[5 rows x 7 columns]

In [0]:
racial_data_1900["Other Colored"] = racial_data_1900["Other Colored >> Male"] + racial_data_1900["Other Colored >> Female"]
racial_data_1900["Negro"] = racial_data_1900["Negro >> Male"] + racial_data_1900["Negro >> Female"]
racial_data_1900 = racial_data_1900.drop(columns=["Other Colored >> Male", "Other Colored >> Female", "Negro >> Male", "Negro >> Female"])

In [86]:
racial_data_1900.head()

YEAR              STATE    White  Other Colored   Negro
0  1900            Alabama  1001152         827545  827307
1  1900   Alaska Territory    30493          33099     168
2  1900  Arizona Territory    92903          30028    1848
3  1900           Arkansas   944580         366984  366856
4  1900         California  1402727          82326   11045

In [0]:
racial_data_territories_1900 = racial_data_1900[racial_data_1900["STATE"].str.contains("Territory")]

In [88]:
racial_data_territories_1900

YEAR                 STATE   White  Other Colored  Negro
1   1900      Alaska Territory   30493          33099    168
2   1900     Arizona Territory   92903          30028   1848
11  1900      Hawaii Territory   66890          87111    233
14  1900      Indian Territory  302680          89380  36853
32  1900  New Mexico Territory  180207          15103   1610
37  1900    Oklahoma Territory  367524          30807  18831

Territories are starting to get sizable black populations, so keeping them.